test/deriver/pp_factory.expected

```diff
@@ -0,0 +1,9 @@
+module A = struct module B = struct type t = char
+                                    let default = 'c' end end
+type simple_record =
+  {
+  int_field: int ;
+  string_field: string ;
+  other_field: A.B.t }[@@deriving factory]
+let factory_simple_record ?(int_field= 0)  ?(string_field= "") 
```

Comment:
Stupid question: would it be worth inlining the default in the output if it exists (so in this case you'd get `?(other_field='c')`?

Reply:
Unfortunately it's not possible. We're deriving values from the type definition only. That means that in the scope of the generator, we don't know what `A.B.default` is, we don't even know if it exists.

The deriver handles the basic types and relies on the user for defining default values for the rest of them.

Does that make sense?